src/main/java/com/winner_is_kungen/tda367/controller/ConnectionController.java

```diff
@@ -22,6 +24,11 @@
 
 		this.fromCP.changeColor(ConnectionPoint.ConnectorColor.DEFAULT_LOW);
 		this.toCP.changeColor(ConnectionPoint.ConnectorColor.DEFAULT_LOW);
+
+		this.setOnMousePressed(mouseEvent -> {
+			mouseEvent.consume();
+			onClick();
+		});
```

Comment:
Why is some of the "onClick" logic here and some in the `onClick` method?

Reply:
mouseEvent.consume() is needed to make sure only the connectionController is handling the event. That logic responsible for receiving the event.

 That way onClick() is only responsible for generating an response to the event.

We could have them together but I felt it was better to separate them.

Reply:
Okey